feat: add vector retrieval and update policy/template

pedrocassalpacheco opened the pull request:
This modifies the module configuration to reflect the proper class name, modifies the AstraDBGraphVectorComponent to support different search algorithms, and fixes the JSON parsing for the index policy property. It also adds a template for Graph RAG.

ogabrielluiz commented on src/backend/base/langflow/components/vectorstores/astradb_graph.py:
```diff
@@ -248,10 +256,17 @@ def _add_documents_to_vector_store(self, vector_store) -> None:
             return "similarity_score_threshold"
         if self.search_type == "MMR (Max Marginal Relevance)":
             return "mmr"
+            return "similarity_score_threshold"
```
There are two returns here.

pedrocassalpacheco replied:
Oi Gabriel - certainly, There is also a regression issue that pre-dates me. Looks like the class name on the __init.py__ differs from the astradbvectorgraphcomponent. Should I change that as well so the unit tests pass?

ogabrielluiz reviewed (changes requested):
Hey @pedrocassalpacheco 

Thanks for the improvements. 
Could you review the PR title to conform with conventional commits (https://www.conventionalcommits.org/en/v1.0.0/#summary) ?

pedrocassalpacheco commented:
> Hey @pedrocassalpacheco
> 
> Thanks for the improvements. Could you review the PR title to conform with conventional commits (https://www.conventionalcommits.org/en/v1.0.0/#summary) ?

@ogabrielluiz - can you review this again? I think I addressed your concerns ...

pedrocassalpacheco commented:
@ogabrielluiz - I removed the template from the PR as requested
